cmd/firectl/main.go

```diff
@@ -181,26 +197,85 @@ func main() {
 		Debug:             opts.Debug,
 	}
 
-	m, err := firecracker.NewMachine(fcCfg, firecracker.WithLogger(log.NewEntry(logger)))
+	ctx := context.Background()
+	cmd, cleanup, err := buildCommand(ctx, fcCfg.SocketPath, opts)
```

Comment:
Why do we want this outside `firecracker` package?
And if we do really want, consider approach similar to `FirecrackerClient` (add something like `WithFirecrackerProcessRunner` opt and fallback implementation, if not provided).

Reply:
Are you asking why do we want the `buildCommand` to be outside of the `firecracker` package?

Reply:
Why do we force user to manage `Command` stuff.

Reply:
Ah, because it is more extensible. The goal for the command builder was to make it easy for users to deal with it.

Users can now specify how they want to start the VMM, which is more powerful than doing it all under the hood. If we want, we can add a `DefaultFirecrackerCommand` that is a prebuild command that users can use. A good example of a user wanting to extend may be they want to remote start a command via SSH. The prior implementation made that impossible to do.

Reply:
`firecracker.NewMachine(cfg, firecracker.WithLogger(), firecracker.WithProcessRunner(cmd))`

Reply:
It makes sense to provide default implementation as well.

Reply:
Yea, `WithProcessRunner` sounds like a good idea. If one isn't provided use the default one. What do you think?

Reply:
Yes!